### PR TITLE
Add Rx, Ry, Rz gates

### PR DESCRIPTION
**Pull request by tanujkhattar**
This PR creates separate `Rx`, `Ry`, `Rz` gates.

### CODE REVIEW

File: cirq/ops/common_gates.py [hunk 1]
```diff
@@ -279,6 +269,50 @@ def __repr__(self) -> str:
...
+    The unitary matrix of ``Rx(rads=t)`` is:
+
+    exp(-i X t/2) =  [ cos(t/2)  -isin(t/2)]
+                     [-isin(t/2)  cos(t/2) ]
+
+    The gate corresponds to the traditionally defined rotation matrices about the Pauli X axis.
+    """
+
+    def __init__(self, *, rads: value.TParamVal):
+        super(Rx, self).__init__(exponent=rads / _pi(rads), global_shift=-0.5)
```
[the remark this answers is not in the document]
Comment (tanujkhattar) [reply cirq/ops/common_gates.py]: done

File: cirq/ops/common_gates.py [hunk 2]
```diff
@@ -279,6 +269,50 @@ def __repr__(self) -> str:
...
+
+    def _with_exponent(self: 'Rx', exponent: value.TParamVal) -> 'Rx':
+        return type(self)(rads=exponent * _pi(exponent))
```
[the remark this answers is not in the document]
Comment (tanujkhattar) [reply cirq/ops/common_gates.py]: nope, fixed.

File: cirq/ops/common_gates.py [hunk 3]
```diff
@@ -279,6 +269,50 @@ def __repr__(self) -> str:
...
+
+    The unitary matrix of ``Rx(rads=t)`` is:
+
+    exp(-i X t/2) =  [ cos(t/2)  -isin(t/2)]
+                     [-isin(t/2)  cos(t/2) ]
+
+    The gate corresponds to the traditionally defined rotation matrices about the Pauli X axis.
+    """
+
+    def __init__(self, *, rads: value.TParamVal):
```
[the remark this answers is not in the document]
Comment (tanujkhattar) [reply cirq/ops/common_gates.py:283]: done.

File: cirq/ops/common_gates.py [hunk 4]
```diff
@@ -216,25 +220,20 @@ def _pauli_expansion_(self) -> value.LinearDict[str]:
...
-            return f'Rx({angle_str})'
-
         return protocols.CircuitDiagramInfo(
             wire_symbols=('X',), exponent=self._diagram_exponent(args)
         )
 
     def _qasm_(self, args: 'cirq.QasmArgs', qubits: Tuple['cirq.Qid', ...]) -> Optional[str]:
         args.validate_version('2.0')
-        if self._exponent == 1:
+        if self._exponent == 1 and self._global_shift != -0.5:
```
[the remark this answers is not in the document]
Comment (tanujkhattar) [reply cirq/ops/common_gates.py:229]: This is to fix (at least partially) #1800.

[the remark this answers is not in the document]
Comment (maffoo) [reply cirq/ops/common_gates.py:229]: Makes sense. Thanks.

**Review by maffoo — APPROVED** [review]
LGTM. Might be good to get another maintainer to look it over as well, since these are such important gates, but up to you.

File: cirq/ops/__init__.py [hunk 5]
```diff
@@ -58,6 +58,9 @@
     CZPowGate,
     H,
     HPowGate,
+    Rx,
+    Ry,
+    Rz,
     rx,
```
Comment (balopat) [inline cirq/ops/__init__.py:64]: I think we can deprecate these in favor for the R* constructors.

**Comment by maffoo** [comment]
Discussed in cirq sync: we should implement `__pow__` on these `R*` gates so they stay as rotation gates when exponentiating.

File: cirq/ops/__init__.py (hunk 5 above)
Comment (tanujkhattar) [reply cirq/ops/__init__.py:64]: Filed an issue to track this: #3929

**Comment by tanujkhattar** [comment]
@maffoo Exponentiating R* gates work as expected because `_with_exponent` is already overloaded. Added explicit tests for the same.

**Comment by tanujkhattar** [comment]
@balopat Do you want to take another look or should I merge?

**Review by balopat — APPROVED** [review]

**Comment by mpharrigan** [comment]
This breaks pytket which uses `type(obj)` to look up gate types instead of doing an `isinstance()` check. Ironically the pytket name for their rotation gate it `Rx` etc. but they are only looking for the `PowGate`s in cirq circuits. (pytket 0.7)

**Pull request merged**